torch/testing/_internal/common_utils.py

```diff
@@ -1999,6 +1999,7 @@ def wrap_with_cuda_memory_check(self, method):
     # show up red).
     # Otherwise, the flaky test will show up green while its stats are captured by test reports.
     def _run_with_retry(self, result=None, num_runs_left=0, report_only=True, num_red=0, num_green=0):
+        warnings.filterwarnings('ignore', 'TypedStorage is deprecated')
```

Comment:
@ezyang, do you think it's alright to do this? It does suppress the warnings from tests, at least locally for me. I'll double check the CI logs after they finish

Reply:
ok. But there's also something else I'd like you to investigate, which is why the warnings keep popping up repeatedly; they should only trigger once per process and then not again.

Reply:
So it looks like `test_proxy_tensor.py`, with `PYTORCH_TEST_WITH_CROSSREF=1` has a few places where different warnings (not just the TypedStorage deprecation warning) are getting duplicated. Here's one example:

<details><summary>click to expand</summary>

```
$ PYTORCH_TEST_WITH_CROSSREF=1 python test/test_proxy_tensor.py -k test_make_fx_fake_exhaustive_nn_functional_upsample_nearest_cpu_float32
/work2/kurtamohler/development/pytorch-0/torch/backends/cudnn/__init__.py:91: UserWarning: PyTorch was compiled without cuDNN/MIOpen support. To use cuDNN/MIOpen, rebuild PyTorch making sure the library is visible to the build system.
  warnings.warn(
/work2/kurtamohler/development/pytorch-0/test/test_proxy_tensor.py:124: UserWarning: Couldn't import torchvision. Some of our tests use it, try to install it with commands from pytorch.org, post-fixed with `--no-deps` to avoid overwriting the pytorch installation
  warnings.warn("Couldn't import torchvision. Some of our tests use it, try "
/work2/kurtamohler/development/pytorch-0/torch/nn/functional.py:4014: UserWarning: nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.
  warnings.warn("nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.")
/work2/kurtamohler/development/pytorch-0/torch/nn/functional.py:4014: UserWarning: nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.
  warnings.warn("nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.")
/work2/kurtamohler/development/pytorch-0/torch/nn/functional.py:4014: UserWarning: nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.
  warnings.warn("nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.")
/work2/kurtamohler/development/pytorch-0/torch/nn/functional.py:4014: UserWarning: nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.
  warnings.warn("nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.")
/work2/kurtamohler/development/pytorch-0/torch/nn/functional.py:4014: UserWarning: nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.
  warnings.warn("nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.")
/work2/kurtamohler/development/pytorch-0/torch/nn/functional.py:4014: UserWarning: nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.
  warnings.warn("nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.")
/work2/kurtamohler/development/pytorch-0/torch/nn/functional.py:4014: UserWarning: nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.
  warnings.warn("nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.")
/work2/kurtamohler/development/pytorch-0/torch/nn/functional.py:4014: UserWarning: nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.
  warnings.warn("nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.")
/work2/kurtamohler/development/pytorch-0/torch/nn/functional.py:4014: UserWarning: nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.
  warnings.warn("nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.")
/work2/kurtamohler/development/pytorch-0/torch/nn/functional.py:4014: UserWarning: nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.
  warnings.warn("nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.")
/work2/kurtamohler/development/pytorch-0/torch/nn/functional.py:4014: UserWarning: nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.
  warnings.warn("nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.")
/work2/kurtamohler/development/pytorch-0/torch/nn/functional.py:4014: UserWarning: nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.
  warnings.warn("nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.")
/work2/kurtamohler/development/pytorch-0/torch/nn/functional.py:4014: UserWarning: nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.
  warnings.warn("nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.")
/work2/kurtamohler/development/pytorch-0/torch/nn/functional.py:4014: UserWarning: nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.
  warnings.warn("nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.")
/work2/kurtamohler/development/pytorch-0/torch/nn/functional.py:4014: UserWarning: nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.
  warnings.warn("nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.")
/work2/kurtamohler/development/pytorch-0/torch/nn/functional.py:4014: UserWarning: nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.
  warnings.warn("nn.functional.upsample_nearest is deprecated. Use nn.functional.interpolate instead.")
.
----------------------------------------------------------------------
Ran 1 test in 0.080s

OK
```

</details>

I'm not sure why this happens yet, but I was able to fix the places where the TypedStorage warning was being emitted (at least locally--we'll see what happens in CI) by correcting the `handle_torch_function` call in the definition of `Tensor._storage` in  `tools/autograd/templates/python_variable_methods.cpp`. It was `return handle_torch_function(self, "storage");`, but should be `return handle_torch_function(self, "_storage");` instead.

I think it probably makes sense to look into the crossref warning duplication in another issue